### PR TITLE
Do some player operations by Discord id

### PR DESCRIPTION
**Pull request by gausie**
- **Pull out process of identifying a player from an identifier and use across clan functions**
- **Apply formatting**

### CODE REVIEW

File: packages/oaf/src/commands/_player.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  } else {
+    playerIdentifier = input;
+  }
+
+  if (
+    typeof playerIdentifier === "string" &&
+    !validPlayerIdentifier(playerIdentifier)
+  ) {
+    return "Come now, you know that isn't a player. Can't believe you'd try and trick me like this. After all we've been through? 😔";
+  }
```
Comment (horrible-little-slime) [inline packages/oaf/src/commands/_player.ts:48]: I found this section a little hard to read--I got there, but I had to think it through. Would it make more sense to structure this to return early if it's a discord account?

**Review by horrible-little-slime — COMMENTED** [review]
Other than getting a little bamboozled by one section of the code, LGTM

File: packages/oaf/src/commands/_player.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export async function identifyPlayer(input: string) {
+  // If the input is a Discord mention, we'll try to set a player identifier there. Otherwise this just gets set to
+  // the same value as input.
+  let playerIdentifier;
+
+  // Whatever happens we'll try to ascertain whether this is a known player. Because we either do so at the start or
+  // at the end, declare a null variable here.
+  let knownPlayer: Awaited<ReturnType<typeof findPlayer>> = null;
```
Comment (horrible-little-slime) [inline packages/oaf/src/commands/_player.ts:28]: is it worth naming this type, or at least naming `ReturnType<typeof findPlayer>`

File: packages/oaf/src/commands/_player.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import type { Prisma } from "@prisma/client";
+
+import { prisma } from "../clients/database.js";
+import { kolClient } from "../clients/kol.js";
+
+export function validPlayerIdentifier(identifier: string) {
+  // If a player id: a number!
+  // If a username: 3 to 30 alphanumeric characters, starting with alpha, may contain underscores or spaces
+  return /^([a-zA-Z][a-zA-Z0-9_ ]{2,29})|[0-9]+$/.test(identifier);
```
Comment (Semenar) [inline packages/oaf/src/commands/_player.ts:9]: Would it be good to split the regex into several for readability?
Also it does not seem to allow Discord handles which the code below parses.

**Comment by horrible-little-slime** [comment]
So in retrospect I'm not actually sure these changes are good; @gausie either revert my last few changes and merge, or merge as is. Or a secret third thing! Dealer's choice!

**Pull request merged**